internal/cache/util/util.go

```diff
@@ -99,3 +101,25 @@ func IsCacheHandleInvalid(readErr error) bool {
 		strings.Contains(readErr.Error(), ErrInSeekingFileHandleMsg) ||
 		strings.Contains(readErr.Error(), ErrInReadingFileHandleMsg)
 }
+
+func CheckCacheDirectoryPermissions(dirPath string) (bool, error) {
+	_, statErr := os.Stat(dirPath)
+
+	if statErr != nil {
```

Comment:
Should we check that stat error is particularly because the directory does not exists?

Reply:
I see that statErr only has error string, dont think its good idea to comapre error string as we dont want to ti with hard code string value. `[ref](https://stackoverflow.com/questions/42035104/how-to-unit-test-go-errors)`

Reply:
Can we compare it with syscall.ENOENT?